src/interchange/src/protobuf.rs

```diff
@@ -91,24 +63,35 @@ impl DecodedDescriptors {
     }
 }
 
+/// Configures a [`Decoder`].
+#[derive(Debug)]
+pub struct DecoderConfig {
```

Comment:
As you probably have learned by now (lol), I'm all for more structs but I don't think I follow the value of adding a separate `DecoderConfig` with the way it's currently used.  It looks like it's just created to pass args into `Decoder::new`, which then immediately splits them back out to store internally

I think if you'd like to have it, then we ought to change `Decoder` to look like this:
```
pub struct Decoder {
    config: DecoderConfig
    packer: Row,
}
```
Otherwise, I think keeping the arguments separate in the `Decoder` constructor is simpler.

What do you think?

Reply:
So the pattern here is the general pattern we use in a few other places for essentially providing named options to structs. See, e.g.:

* https://github.com/MaterializeInc/materialize/blob/6bb2a5c61b3ac112f453361a124f978b53780e0f/src/materialized/src/lib.rs#L225-L234
* https://github.com/MaterializeInc/materialize/blob/6bb2a5c61b3ac112f453361a124f978b53780e0f/src/materialized/src/lib.rs#L237-L250
* https://github.com/MaterializeInc/materialize/blob/6bb2a5c61b3ac112f453361a124f978b53780e0f/src/materialized/src/lib.rs#L277-L288

It feels a little [duplicative](https://github.com/MaterializeInc/materialize/blob/6bb2a5c61b3ac112f453361a124f978b53780e0f/src/materialized/src/http.rs#L78-L95) for sure in isolation, but it makes space for situations where you need to munge a config item during initialization:

https://github.com/MaterializeInc/materialize/blob/6bb2a5c61b3ac112f453361a124f978b53780e0f/src/pgwire/src/server.rs#L75

As soon as you need to munge _one_ item in a config struct, you can no longer wholesale embed the config struct in the resulting type.

I agree it's not super worth it for a single bool param, so I just dropped the commit. It becomes increasingly important as you add params though, with something like the coordinator being a situation where you 100% need it if you want to have any chance of understanding the call site.

Reply:
Ahh I see -- that pattern makes sense